src/main/java/seedu/address/logic/parser/task/DeleteTaskCommandParser.java

```diff
@@ -0,0 +1,28 @@
+package seedu.address.logic.parser.task;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.task.DeleteTaskCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.TaskParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteTaskCommand object
+ */
+public class DeleteTaskCommandParser implements Parser<DeleteTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteCommand
+     * and returns an DeleteCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteTaskCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+
+        Index index = TaskParserUtil.parseIndex(argMultimap.getPreamble());
```

Comment:
In other commands, we caught the ParseException from parseIndex and throw a new ParseException containing String.format(MESSAGE_INVALID_COMMAND_FORMAT, XXXCommand.MESSAGE_USAGE) that will be displayed to the user. You can consider following the same format for consistency of our UI!